### PR TITLE
Adding support for Segment group in customer.io.

### PR DESCRIPTION
**Pull request by kishoredevarasettyn**
Adding support for Segment group in customer.io. Group is synonymous to Objects in customer.io
New destination action createUpdateObject added to the list and tested end to end using https://app.segment.com/dev-center/actions-tester

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

Added Unit tests with 100% coverage and tested end-to-end using local server.

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/customerio/__tests__/createUpdateObject.test.ts [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+        useDefaultMappings: true
+      })
+
+      expect(responses.length).toBe(1)
+      expect(responses[0].status).toBe(200)
+      expect(responses[0].headers.toJSON()).toMatchObject({
+        'x-customerio-region': 'US',
+        'content-type': 'application/json'
+      })
+      expect(responses[0].data).toMatchObject({})
```
Comment (mike-engel) [inline packages/destination-actions/src/destinations/customerio/__tests__/createUpdateObject.test.ts:48]: You should be asserting the structure of the data going to cio (here, and for the other tests as well). Here's an example: https://github.com/segmentio/action-destinations/blob/d7111d36645f7e45b91996fac2cc0dea15283686/packages/destination-actions/src/destinations/customerio/__tests__/createUpdatePerson.test.ts#L53

File: packages/destination-actions/src/destinations/customerio/createUpdateObject/index.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        'The ID used to uniquely identify an object in Customer.io. [Learn more](https://customer.io/docs/object-relationships).',
+      type: 'string',
+      required: true,
+      default: {
+        '@path': '$.groupId'
+      }
+    },
+    created_at: {
+      label: 'Created At',
+      description: 'A timestamp of when the person was created.',
```
[the remark this answers is not in the document]
Comment (kishoredevarasettyn) [reply packages/destination-actions/src/destinations/customerio/createUpdateObject/index.ts]: corrected

File: packages/destination-actions/src/destinations/customerio/createUpdateObject/index.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        'Optional attributes for the object. When updating an object, attributes are added or updated, not removed.',
+      type: 'object',
+      default: {
+        '@path': '$.traits'
+      }
+    },
+    user_id: {
+      label: 'User ID',
+      description:
+        'The ID used to relate user to an object in Customer.io. [Learn more](https://customer.io/docs/identifying-people/#identifiers).',
```
[the remark this answers is not in the document]
Comment (kishoredevarasettyn) [reply packages/destination-actions/src/destinations/customerio/createUpdateObject/index.ts]: corrected

File: packages/destination-actions/src/destinations/customerio/createUpdateObject/index.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+      type: 'string',
+      default: {
+        '@path': '$.userId'
+      }
+    },
+    anonymous_id: {
+      label: 'Anonymous ID',
+      description:
+        'An anonymous ID for when no Person ID exists. [Learn more](https://customer.io/docs/anonymous-events/).',
```
[the remark this answers is not in the document]
Comment (kishoredevarasettyn) [reply packages/destination-actions/src/destinations/customerio/createUpdateObject/index.ts]: Added description

File: packages/destination-actions/src/destinations/customerio/createUpdateObject/index.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    let typeID
+    if (customAttributes) {
+      typeID = customAttributes.object_type_id
+    }
+
+    if (createdAt) {
+      body.created_at = createdAt
+    }
+    body.type = 'object'
+    body.identifiers = { type_id: typeID ?? '1', id: objectID }
```
[the remark this answers is not in the document]
Comment (kishoredevarasettyn) [reply packages/destination-actions/src/destinations/customerio/createUpdateObject/index.ts:91]: Yes, right now, we support only one object and it should be 1

File: packages/destination-actions/src/destinations/customerio/index.ts [hunk 6]
```diff
@@ -9,6 +9,8 @@ import type { DestinationDefinition } from '@segment/actions-core'
 import type { Settings } from './generated-types'
 import { AccountRegion, trackApiEndpoint } from './utils'
 
+import createUpdateObject from './createUpdateObject'
```
[the remark this answers is not in the document]
Comment (kishoredevarasettyn) [reply packages/destination-actions/src/destinations/customerio/index.ts]: moved above

File: packages/destination-actions/src/destinations/customerio/__tests__/createUpdateObject.test.ts (hunk 1 above)
Comment (kishoredevarasettyn) [reply packages/destination-actions/src/destinations/customerio/__tests__/createUpdateObject.test.ts:49]: Added assets for the json

**Review by mike-engel — APPROVED** [review]

**Review by marinhero — APPROVED** [review]
LGTM. This will go live on Wednesday of next week. Well done!

**Comment by kishoredevarasettyn** [comment]
Hi @marinhero , apologies. made a small change to send traits in attributes property.

**Review by marinhero — APPROVED** [review]

**Pull request merged**